reactivesocket-tck-drivers/src/main/java/io/reactivesocket/tckdrivers/client/JavaClientDriver.java

```diff
@@ -0,0 +1,554 @@
+                rspub.subscribe(rssub);
+                break;
+            case "sub":
+                TestSubscriber<Payload> rsubsub = new TestSubscriber<>(0L);
+                payloadSubscribers.put(args[2], rsubsub);
+                idToType.put(args[2], args[1]);
+                ReactiveSocket rsubclient = createClient.get();
+                Publisher<Payload> rsubpub = rsubclient.requestSubscription(new PayloadImpl(args[3], args[4]));
+                rsubpub.subscribe(rsubsub);
+                break;
+            case "fnf":
+                TestSubscriber<Void> fnfsub = new TestSubscriber<>(0L);
+                fnfSubscribers.put(args[2], fnfsub);
+                idToType.put(args[2], args[1]);
+                ReactiveSocket fnfclient = createClient.get();
+                Publisher<Void> fnfpub = fnfclient.fireAndForget(new PayloadImpl(args[3], args[4]));
+                fnfpub.subscribe(fnfsub);
+                break;
+            default:break;
+        }
+    }
+
+    /**
+     * This function takes in an iterator that is parsing through the test, and collects all the parts that make up
+     * the channel functionality. It then create a thread that runs the test, which we wait to finish before proceeding
+     * with the other tests.
+     * @param args
+     * @param iter
+     * @param name
+     */
+    private void handleChannel(String[] args, Iterator<String> iter, String name) {
+        List<String> commands = new ArrayList<>();
+        String line = iter.next();
+        // channel script should be bounded by curly braces
+        while (!line.equals("}")) {
+            commands.add(line);
+            line = iter.next();
+        }
+        // set the initial payload
+        Payload initialPayload = new PayloadImpl(args[1], args[2]);
+
+        // this is the subscriber that will request data from the server, like all the other test subscribers
+        TestSubscriber<Payload> testsub = new TestSubscriber<>(1L);
+        ParseChannel superpc = null;
+        CountDownLatch c = new CountDownLatch(1);
+
+        // we now create the publisher that the server will subscribe to with its own subscriber
+        // we want to give that subscriber a subscription that the client will use to send data to the server
+        ReactiveSocket client = createClient.get();
+        PCTWrapper mypct = new PCTWrapper();
+        Publisher<Payload> pub = client.requestChannel(new Publisher<Payload>() {
+            @Override
+            public void subscribe(Subscriber<? super Payload> s) {
+                ParseMarble pm = new ParseMarble(s);
+                TestSubscription ts = new TestSubscription(pm, initialPayload, s);
+                s.onSubscribe(ts);
+                ParseChannel pc = new ParseChannel(commands, testsub, pm, name);
+                ParseChannelThread pct = new ParseChannelThread(pc);
+                pct.start();
+                mypct.set(pct);
+                c.countDown();
```

Comment:
It seems you don't wait for the thread to finish?
Also, why using a Thread, could you use synchronous code here?